dwave/plugins/sklearn/nearest_neighbors.pyx

```diff
@@ -0,0 +1,117 @@
+# distutils: language = c++
+import numpy as np
+cimport numpy as np
+from libcpp.vector cimport vector
+
+cdef extern from "math.h":
+    double log(double x) nogil
+
+def calculate_mi(np.ndarray[np.float_t,ndim = 2] X,
```

Comment:
In modern Cython, they want us to use [Typed Memoryviews](https://cython.readthedocs.io/en/latest/src/userguide/memoryviews.html). So for instance here it would be
```suggestion
def calculate_mi(np.float_t[:] X,
```